modules/apps/oauth2-provider/oauth2-provider-client/src/main/resources/META-INF/resources/liferay-oauth2-tester/liferayOauth2Tester.js

```diff
@@ -0,0 +1,609 @@
+document
+	.getElementById('changeModeButton')
+	.addEventListener('click', (event) => {
+		event.preventDefault();
+		if (!pkceMode) {
+			window.location.replace(
+				`${window.location.href.split('?')[0]}?pkce=1`
+			);
+		}
+		else {
+			window.location.replace(window.location.pathname);
+		}
+	});
+
+/*
+ * Send introspect access token request.
+ */
+document
+	.getElementById('introspectAccessTokenButton')
+	.addEventListener('click', async (event) => {
+		event.preventDefault();
+		if (accessToken.value.trim()) {
+			const response = await sendIntrospectRequest(
+				'access_token',
+				accessToken.value.trim()
+			);
+			if (response) {
+				resultTextArea.value = JSON.stringify(response, null, 2);
+				resultTextArea.innerHTML = JSON.stringify(response);
+			}
+		}
+		else {
+			alert('Access token is missing!');
+		}
+	});
+
+/*
+ * Send introspect refresh token request.
+ */
+document
+	.getElementById('introspectRefreshTokenButton')
+	.addEventListener('click', async (event) => {
+		event.preventDefault();
+		if (refreshToken.value.trim()) {
+			const response = await sendIntrospectRequest(
+				'refresh_token',
+				refreshToken.value.trim()
+			);
+			if (response) {
+				resultTextArea.value = JSON.stringify(response, null, 2);
+				resultTextArea.innerHTML = JSON.stringify(response);
+			}
+		}
+		else {
+			alert('Refresh token is missing!');
+		}
+	});
+
+/*
+ * Generate code challenge.
+ */
+document
+	.getElementById('generateCodeChallenge')
+	.addEventListener('click', (event) => {
+		event.preventDefault();
+		const codeChallenge = (document.getElementById(
+			'codeChallenge'
+		).value = random());
+		const codeSHA256 = CryptoJS.SHA256(
+			CryptoJS.enc.Utf8.parse(codeChallenge)
+		);
+		document.getElementById(
+			'convertedCodeChallenge'
+		).value = CryptoJS.enc.Base64url.stringify(codeSHA256);
+		document.getElementById('getAuthorizationCode').disabled = false;
+	});
+
+/*
+ * Refresh the access token.
+ */
+document
+	.getElementById('refreshTokenButton')
+	.addEventListener('click', async (event) => {
+		event.preventDefault();
+		if (refreshToken.value.trim()) {
+			const body = {
+				client_id: clientId.value,
+				grant_type: 'refresh_token',
+				refresh_token: refreshToken.value.trim(),
+			};
+
+			if (!pkceMode) {
+				body.client_secret = clientSecretOrCodeChallenge.value;
+			}
+
+			const response = await sendHttpRequest(
+				tokenUrl(),
+				'POST',
+				'application/x-www-form-urlencoded',
+				body
+			);
+
+			if (response) {
+				accessToken.value = response.access_token;
+				refreshToken.value = response.refresh_token;
+			}
+		}
+		else {
+			alert('Refresh token is missing!');
+		}
+	});
+
+/*
+ * Reset all the forms.
+ */
+document.getElementById('resetButton').addEventListener('click', (event) => {
+	event.preventDefault();
+	let url = window.location.href.split('?')[0];
+	url += pkceMode ? '?pkce=1' : '';
+	window.location.replace(url);
+});
+
+window.addEventListener('load', () => {
+
```

Comment:
remove new line

Reply:
This one is getting added back by prettier. I guess due to the comment on the next line.